File: rb/Irrgarten/LabyrinthCharacter.rb
#encoding:utf-8
module Irrgarten
    class LabyrinthCharacter

        attr_reader :name, :row, :col, :intelligence, :strength, :health

        @@INVALID_POS=-1

        def initialize(name, intelligence, strength, health)
            @name=name
            @intelligence=intelligence
            @strength=strength
            @health=health
            @row=@col=@@INVALID_POS
        end

        def copy (other)
            @name=other.name
            @intelligence=other.intelligence
            @health=other.health
            @row=other.row
            @col=other.col
        end

        def dead?
            return @health==0.0
        end

        def get_row
            @row
        end

        def get_col
            @col
        end

        def set_pos(row,col)
            @row=row
            @col=col
        end

        def to_s
            "#{@name} [I:#{@intelligence}, S:#{@strength}, H:#{@health}, P(#{@row},#{@col})]"
        end

        def attack
        end

        def defend
        end

        protected

        def get_intelligence
            @intelligence
        end

        def get_strength
            @strength
        end

        def get_health
            @health
        end

        def set_health(health)
            @health=health
        end

        def got_wounded
            @health-=1
        end
    end
end

File: ExamenP5/Irrgarten/src/irrgarten/LabyrinthCharacter.java

package irrgarten;

/**
 *
 * @author Sergio Salvador Gil
 */
public abstract class LabyrinthCharacter {
    protected static final int INVALID_POS=-1;
    
    private String name;
    private float intelligence;
    private float strength;
    private float health;
    private int row,col;
    
    public LabyrinthCharacter(String name, float intelligence, float strength, float health){
        this.name=name;
        this.intelligence=intelligence;
        this.strength=strength;
        this.health=health;
        row=col=INVALID_POS;
    }
    
    public LabyrinthCharacter(LabyrinthCharacter other){
        this.name = other.name;
        this.intelligence = other.intelligence;
        this.strength = other.strength;
        this.health = other.health;
        this.row = other.row;
        this.col = other.col;
    }
    
    public boolean dead(){
        return health==0.0f;
    }
    
    public int getRow(){
        return row;
    }
    
    public int getCol(){
        return col;
    }
    
    protected float getIntelligence(){
        return intelligence;
    }

    protected float getStrength() {
        return strength;
    }

    protected float getHealth() {
        return health;
    }
    
    protected void setHealth(float health) {
        this.health = health;
    }
    
    public void setPos(int row, int col){
        this.row=row;
        this.col=col;
    }
    
    @Override
    public String toString(){
        return name + " [i:" + intelligence + ", s:" + strength + ", h:" + health + ", p(" + row + "," + col + ")]";
    }
    
    protected void gotWounded(){
        health-=1;
    }
    
    public abstract float attack();
    public abstract boolean defend(float receivedAttack);
}
